sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecutionSuite.scala

```diff
@@ -68,4 +73,44 @@ class MicroBatchExecutionSuite extends StreamTest with BeforeAndAfter {
       CheckNewAnswer((25, 1), (30, 1))   // This should not throw the error reported in SPARK-24156
     )
   }
+
+  test("SPARK-28597: Add config to retry spark streaming's meta log when it met") {
+    val s = MemoryStream[Int]
+    val df = s.toDF()
+    // Specified checkpointLocation manually to init metadata file
+    val tmp =
+      new File(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString).getCanonicalPath
+    testStream(s.toDF())(
+      StartStream(checkpointLocation = tmp)
+    )
+
+    // fail with less retries
+    df.sparkSession.sessionState.conf.setConfString(
+      SQLConf.STREAMING_CHECKPOINT_FILE_MANAGER_CLASS.parent.key,
+      classOf[FakeFileSystemBasedCheckpointFileManager].getName)
```

Comment:
In the test suite, we use `withSQLConf` instead of `setConfString` and `unsetConf` combination.
Please refer `CheckpointFileManagerSuite` for the usage of `withSQLConf`.